internal/tools/terraformlinter/terraform_linter_test.go

```diff
@@ -0,0 +1,662 @@
+		disable_on_destroy = true
+	}
+	`
+	testProjectOutOfOrder := `
+	resource "google_project_service" "run_api" {  
+		service = "run.googleapis.com"  
+		project = "some_project_id"
+		disable_on_destroy = true
+	}
+	`
+	testDependsOnCorrect := `
+	resource "google_project_service" "run_api" {  
+		service = "run.googleapis.com"  
+		disable_on_destroy = true
+		depends_on = [
+			"something"
+		]
+	}
+	`
+	testDependsOnOutOfOrder := `
+	resource "google_project_service" "run_api" {  
+		service = "run.googleapis.com"  
+		depends_on = [
+			"something"
+		]
+		disable_on_destroy = true
+	}
+	`
+	testLifecycleCorrect := `
+	resource "google_project_service" "run_api" {  
+		service = "run.googleapis.com"  
+		disable_on_destroy = true
+		lifecycle = {
+			prevent_destroy = true
+		}
+	}
+	`
+	testLifecycleOutOfOrder := `
+	resource "google_project_service" "run_api" {  
+		service = "run.googleapis.com"  
+		lifecycle = {
+			prevent_destroy = true
+		}
+		disable_on_destroy = true
+	}
+	`
+	testTrailingMixCorrect := `
+	resource "google_project_service" "run_api" {  
+		service = "run.googleapis.com"  
+		disable_on_destroy = true
+		depends_on = [
+			"something"
+		]
+		lifecycle = {
+			prevent_destroy = true
+		}
+	}
+	`
+	testTrailingMixOutOfOrder := `
+	resource "google_project_service" "run_api" {  
+		service = "run.googleapis.com"  
+		disable_on_destroy = true
+		lifecycle = {
+			prevent_destroy = true
+		}
+		depends_on = [
+			"something"
+		]
+	}
+	`
+	testSourceCorrect := `
+	resource "google_project_service" "run_api" {  
+		source = "http://somerepo"
+
+		service = "run.googleapis.com"  
+		disable_on_destroy = true
+	}
+	`
+	testSourceMissingNewLine := `
+	resource "google_project_service" "run_api" {  
+		source = "http://somerepo"
+		service = "run.googleapis.com"  
+		disable_on_destroy = true
+	}
+	`
+	testSourceOutOfOrder := `
+	resource "google_project_service" "run_api" {  
+		service = "run.googleapis.com"  
+		source = "http://somerepo"
+		disable_on_destroy = true
+	}
+	`
+	testAllCorrect := `
+	resource "google_project_service" "run_api" {  
+		for_each = toset(["name"])
+		provider = "someprovider"
+
+		project = "pid"
+		project_id = "pid"
+		folder = "fid"
+		organization = "abcxyz"
+
+		service = "run.googleapis.com"  
+		disable_on_destroy = true
+
+		depends_on = [
+			"something"
+		]
+		lifecycle = {
+			prevent_destroy = true
+		}
+	}
+	`
+	testMixedOutOfOrder := `
+	resource "google_project_service" "run_api" {  
+		folder = "fid"
+		provider = "someprovider"
+		project = "pid"
+		for_each = toset(["name"])
+		project_id = "pid"
+		service = "run.googleapis.com"  
+		lifecycle = {
+			prevent_destroy = true
+		}
+		organization = "abcxyz"
+		disable_on_destroy = true
+		depends_on = [
+			"something"
+		]
+	}
+	`
+
+	cases := []struct {
+		name      string
+		content   string
+		filename  string
+		expect    []*ViolationInstance
+		wantError bool
+	}{
+		{
+			name:      "No special attributes",
```

Comment:
Oops, this diff was suppressed by GitHub.

1. `name` attributes should prefer lowercase and have no spaces (use underscores). It helps with grepping for failed tests.

1. I think we should inline the `content` so the test is fully self-contained.